pkg/runner/cluster_k8s.go

```diff
@@ -273,4 +243,146 @@ func (*ClusterK8sRunner) CompatibleBuilders() []string {
 	return []string{"docker:go"}
 }
 
-func int32Ptr(i int32) *int32 { return &i }
+func getPodLogs(clientset *kubernetes.Clientset, podName string) string {
+	podLogOpts := v1.PodLogOptions{
+		TailLines: int64Ptr(2),
+	}
+	req := clientset.CoreV1().Pods("default").GetLogs(podName, &podLogOpts)
+	podLogs, err := req.Stream()
+	if err != nil {
+		return "error in opening stream"
+	}
+	defer podLogs.Close()
+
+	buf := new(bytes.Buffer)
+	_, err = io.Copy(buf, podLogs)
+	if err != nil {
+		return "error in copy information from podLogs to buf"
+	}
+	str := buf.String()
+
+	return str
+}
+
+func monitorTestplanRunState(ctx context.Context, pool *pool, log *zap.SugaredLogger, runID string, k8sNamespace string, replicas int) error {
+	client, err := pool.Acquire(ctx)
+	if err != nil {
+		return err
+	}
+	defer pool.Release(client)
+
+	start := time.Now()
+	for {
+		if time.Since(start) > 10*time.Minute {
```

Comment:
I think we'll want to extract these timing assumptions to runner config variables.